File: Typescript/Practice/26072022/medium/medium_findNumberDigits.ts
#!/usr/bin/env npx ts-node

function num_of_digits(num: number): number {
  let digit: number = 0;
  if (num === 0) {
    return digit;
  } else {
    while (true) {
      if (Math.floor(num / (10 ** digit)) !== 0) {
        digit++;
        continue;
      };
      break;
    };
    return digit;
  };
};

console.log(num_of_digits(1000));
console.log(num_of_digits(12));
console.log(num_of_digits(1305981031));
console.log(num_of_digits(0));

File: Typescript/Practice/26072022/medium/medium_findNumberDigits.py
#!/usr/bin/env python

def num_of_digits(num):
  digits = 0
  if num == 0:
    digits = 1
  else:
    while True:
      if num // (10 ** digits) != 0:
        digits += 1
        continue
      break
  return digits

print(num_of_digits(1000))
print(num_of_digits(12))
print(num_of_digits(1305981031))
print(num_of_digits(0))

